src/main/kotlin/io/github/yuutoproject/yuutobot/commands/minigame/MinigameInstance.kt

```diff
@@ -0,0 +1,184 @@
+        try {
+            currentQuestion = questions.removeAt(0)
+        } catch (e: IndexOutOfBoundsException) {
+            endGame(event)
+            return
+        }
+
+        currentAnswers = currentQuestion.answers.map { it.toLowerCase() }
+            .toMutableList()
+
+        if (currentQuestion.type == "FILL") {
+            event.channel.sendMessage(currentQuestion.question).queue()
+        } else if (currentQuestion.type == "MULTIPLE") {
+            val questionString = "${currentQuestion.question}\n"
+
+            val answerString = (currentQuestion.wrong + currentQuestion.answers).shuffled()
+                .mapIndexed { i, answer ->
+                    if (currentAnswers.contains(answer.toLowerCase())) {
+                        currentAnswers.add((i + 1).toString())
+                    }
+
+                    "${i + 1}) $answer"
+                }.joinToString("\n")
+
+            event.channel.sendMessage(questionString + answerString).queue()
+        }
+    }
+
+    private fun endGame(event: GuildMessageReceivedEvent) {
+        // Sort by descending value and then map each value to a line in the scoreboard, then join it
+        val scoreboard = players.entries.sortedByDescending { it.value }.mapIndexed { i, entry ->
+            // If JDA can't find the user, then the game will stop. This shouldn't happen
+            val user = event.jda.getUserById(entry.key)!!
+            "${i + 1}) ${user.asMention} with ${entry.value} points"
+        }.joinToString("\n")
+
+        val embed = EmbedBuilder()
+            .setColor(Color.BLUE)
+            .setTitle("Minigame ended!")
+            .setDescription("Total points:\n$scoreboard")
+        event.channel.sendMessage(embed.build()).queue()
+
+        manager.unregister(this)
+    }
+
+    fun answerReceived(event: GuildMessageReceivedEvent) {
+        // A new guess is made, so we reset the stale-game timer
+        timer = System.currentTimeMillis()
+
+        if (currentAnswers.contains(event.message.contentStripped.toLowerCase())) {
+            players[event.author.id] = players[event.author.id]!! + 1
+            event.channel.sendMessage("${event.author.name} got the point!").queue()
+            rounds += 1
+            progress(event)
+        }
+    }
+
+    fun reactionRecv(event: GuildMessageReactionAddEvent) {
+        if (
+            event.user.isBot ||
+            players.contains(event.user.id) ||
+            begun ||
+            event.messageId != startingMessageID
+        ) return
+
+        players[event.user.id] = 0
+    }
+
+    fun reactionRetr(event: GuildMessageReactionRemoveEvent) {
+        if (
+            event.messageId == startingMessageID &&
+            !begun
+        ) {
+            players.remove(event.user!!.id)
+        }
+    }
+
+    private fun getPlayers(event: GuildMessageReceivedEvent) = if (players.isNotEmpty()) {
+        players.keys.joinToString(", ") { event.jda.getUserById(it)!!.asMention }
```

Comment:
```suggestion
        players.keys.joinToString(", ") { "<@$id>" }
```
You're welcome

Reply:
btw `getUserById` is unreliable due to caching, it might actually be null if we change how the bot caches members